kafka-client/src/main/java/dev/responsive/kafka/store/SizeTrackingBuffer.java

```diff
@@ -0,0 +1,38 @@
+package dev.responsive.kafka.store;
+
+import dev.responsive.model.Result;
+import java.util.Collections;
+import java.util.NavigableMap;
+import java.util.Objects;
+import java.util.TreeMap;
+
+public class SizeTrackingBuffer<K> {
+  private final NavigableMap<K, Result<K>> buffer;
+  private final NavigableMap<K, Result<K>> reader;
+  private final BufferPlugin<K> plugin;
+  private long bytes = 0;
+
+  public SizeTrackingBuffer(final BufferPlugin<K> plugin) {
+    this.plugin = Objects.requireNonNull(plugin);
+    buffer = new TreeMap<>(plugin);
+    reader = Collections.unmodifiableNavigableMap(buffer);
+  }
+
+  public long getBytes() {
+    return bytes;
+  }
+
+  public void put(final K key, final Result<K> value) {
+    bytes += plugin.bytes(key).get().length + (value.isTombstone ? 0 : value.value.length);
```

Comment:
if this is a replace we should subtract the previous element, right?

Reply:
Also if it's a tombstone we don't just add `0`, we should subtract the old size since this is a delete

Reply:
I guess that's basically what Almog was saying

Reply:
🤦 thanks guys. Fixed now.